shared/js/dialer/keypad.js

```diff
@@ -504,6 +516,139 @@ var KeypadManager = {
     }
   },
 
+  /**
+   * Returns true if the number is a speed dial code as described in
+   * 3GPP TS 22.030 6.6.4. Speed dial codes are in the N(N)(N)# format.
+   */
+  _isSpeedDialNumber: function(number) {
+    return !!number.match(/^[0-9][0-9]{0,2}\#$/);
+  },
+
+  /**
+   * Returns the telephony number corresponding to the specified index. Speed
+   * dial numbers are retrieved from the SIM contacts list and not from the
+   * regular contacts.
+   *
+   * @param {Integer} index The index of the speed dial number.
+   * @returns {Promise} A promise that resolves to the corresponding number.
+   */
+  _getSpeedDialNumber: function(index) {
+    var self = this;
+    var cardIndex;
+
+    index--; // Speed dial indexes are 1-based
+
```

Comment:
If we allow user to enter 0# then, this will be a -1.  I didn't get any crash when I tried this, but not sure what else could break here.